### PR TITLE
Docket Switch | Move granted request issues to new appeal stream

### LINKED ISSUE #15098: Intake | Docket Switch | Issues switched to new appeal stream
## User or job story
As a BVA Attorney at the Office of the Clerk of the Board, when I select issues to be switched, they should get copied to a new appeals stream and previous issues should get closed so that there aren't any duplicate issues.

## Acceptance criteria
- [ ] Please put this work behind the feature toggle: `docket_switch`
- [ ] This feature should be accessible to the following user groups: `clerk of the board`
- [ ] When a docket switch is granted or partially granted, create a new appeal stream with type "original"
- [ ] Save the new appeal stream ID on the Docket Switch
- [ ] Copy the granted request issues to the new appeal stream
- [ ] Close the issues on the original stream with "docket_switch"
- [ ] For full grants, the appeal status on the original original stream should be "archived" (this status needs to be added)
- [ ] Ensure that we can navigate to the docket switch from the new appeal stream through a Rails association and vice versa

### Out of scope
- [ ] Updating the front-end label for the appeal stream that shows "original (switched)" when the appeal stream is a result of a docket switch.

### Designs

19 6 Switch Dockets_ Review request - Grant partial issues

### PR DESCRIPTION
**Pull request by leikkisa**
Resolves #15098

### Description
This PR creates a new appeal stream for granted docket switches, and moves the granted issues to the new appeal stream.

## Acceptance criteria
- [x] When a docket switch is granted or partially granted, create a new appeal stream with type "original"
- [x] Save the new appeal stream ID on the Docket Switch
- [x] Copy the granted request issues to the new appeal stream
- [x] Close the issues on the original stream with "docket_switch"
- [x] For full grants, the appeal status on the original original stream should be "archived" (this status needs to be added)
- [x] Ensure that we can navigate to the docket switch from the new appeal stream through a Rails association and vice versa

### Testing Plan
1. Model tests (backend ticket only)

### CODE REVIEW

File: app/models/docket_switch.rb [hunk 1]
```diff
@@ -20,19 +20,36 @@ class DocketSwitch < CaseflowRecord
 
   def process!
     process_denial! if denied?
+    process_granted! if task.is_a?(DocketSwitchGrantedTask)
```
Comment (leikkisa) [inline app/models/docket_switch.rb:23]: covers both granted and partially_granted

File: app/models/docket_switch.rb [hunk 2]
```diff
@@ -18,21 +18,42 @@ class DocketSwitch < CaseflowRecord
     denied: "denied"
   }
 
+  scope :updated_since_for_appeals, lambda { |since|
+    select(:old_docket_stream_id).where("#{table_name}.updated_at >= ?", since)
```
Comment (Sjones352) [inline app/models/docket_switch.rb:22]: @leikkisa What is this method doing?

**Review by Sjones352 — APPROVED** [review]
LGTM

File: app/models/docket_switch.rb (hunk 2 above)
Comment (leikkisa) [reply app/models/docket_switch.rb:22]: There's something that tracks when things connected to appeals get updated, so it triggered a failing test when I didn't include this.

**Pull request merged**